### ISSUE TITLE
Editor doesn't update EML packageId

### ISSUE THREAD

**Issue by csjx** [NCEAS/metacatui#630]
When editing an existing EML document created by the `1.x` Data Registry editor on https://beta.arcticdata.io, the editor doesn't update the `/eml/@packageId` field. Notice the reported `identifier` field mismatch in the `MetadataView`:
![id-mismatch](https://user-images.githubusercontent.com/1437565/40275062-866334e0-5ba2-11e8-8ee3-970420a9927e.png)
Change this field in the DOM when the `SystemMetadata.identifier` changes.  Also, note that the Data Registry (really Metacat) sets the EML document's `SystemMetadata.fileName` to `{pid}.xml`.  We should consider doing something similarly (filename-safe), and also update this field as the EML `pid` changes.

**Comment by csjx** [comment]
Note that the R tools set `SystemMetadata.fileName` to `science_metadata.xml`. While this works, it still seems a little generic. This probably needs discussion.

**Comment by amoeba** [comment]
I think `{pid}.xml` is kinda the least useful thing we could do. `metadata.xml` and `science_metadata.xml` is a little more thoughtful. What about a truncated & sanitized `{/eml/dataset/title/text()}.xml`?

I also think it'd be nice to have uniformity across tools where possible.

**Comment by csjx** [comment]
Good points @amoeba. Yeah, `{pid}.xml` is kind of too minimalistic.  The old Metacat API would take the `/eml/dataset/physical/bjectName`, sanitize it, and then concatenate the `docid`, the concatenate the extension as the last suffix.  Matt made the point (years ago I think) that human readable filenames are good, but that human-readable and unique file names are better. I like your title idea too.

**Comment by amoeba** [comment]
Right on. For web downloads of EML, the user ends up with a file in their
download folder and seeing a random xml file with an identifier they've
never seen before (and really shouldn't have) is odd. I figure the science
metadata title is probably the most familiar thing for them.

**Comment by laurenwalker** [comment]
The new editor sets the metadata file name to the EML title.

**Comment by csjx** [comment]
@laurenwalker, Perhaps it's setting it that way if the file name isn't already set? See the screenshot above, edited by the 2.0 editor, but the file name is still the older pid-based file name. Perhaps we should just set it no matter what, particularly if the title has changed, so the new file name reflects the new title.

**Comment by laurenwalker** [comment]
Yeah, it only sets it when the metadata is first created. I think it's fair to change it each time it's updated.
